code/modules/hydroponics/grown/root.dm

```diff
@@ -28,10 +28,15 @@
 /obj/item/food/grown/carrot/attackby(obj/item/I, mob/user, params)
 	if(I.get_sharpness())
 		to_chat(user, span_notice("You sharpen the carrot into a shiv with [I]."))
-		var/obj/item/knife/shiv/carrot/Shiv = new /obj/item/knife/shiv/carrot
+		var/carrot_blade
+		var/carrot_sword_chance = 100 * (max(0, seed.potency - 50) / 50)
```

Comment:
You've made this a 100% chance of getting a sword at 100 potency, I'm not sure if that's what you were going for

![image](https://github.com/tgstation/tgstation/assets/51863163/7c4657bf-86f7-43dd-bb71-08d706292b09)

Reply:
As an aside, reminder that you get **10**, 100 potency carrots per harvest, not just 1. 

You should slash any probability by 10 to accommodate for that.

Reply:
alright, cut the chance